import{_ as r,M as d,p as n,q as s,R as e,t,N as o,a1 as i}from"./framework-5866ffd3.js";const l={},c=e("h1",{id:"import-and-export",tabindex:"-1"},[e("a",{class:"header-anchor",href:"#import-and-export","aria-hidden":"true"},"#"),t(" Import and export")],-1),h=e("p",null,[e("em",null,"Many projects start with imports, and target exports. Here is where you find what is possible in TaxonWorks.")],-1),p=e("h2",{id:"import",tabindex:"-1"},[e("a",{class:"header-anchor",href:"#import","aria-hidden":"true"},"#"),t(" Import")],-1),u={href:"https://speciesfilegroup.org/events.html",target:"_blank",rel:"noopener noreferrer"},m=i('<h3 id="data-types" tabindex="-1"><a class="header-anchor" href="#data-types" aria-hidden="true">#</a> Data types</h3><p>For bulk/batch uploads, TW supports import of various types of data including checklists, occurrence specimen record data, sequence data, source information (i.e. publications) and more. And using the DwC Archive Import Task, TW offers the use of the current gold standard for sharing specimen occurrence data via a tab-separated text file. This file can stand alone or be part of a Darwin Core Archive (DwCA) zipped file. These (your) data are mapped to Darwin Core terms, TaxonWorks specific elements, or your own custom Data Attributes (details below). Support for importing observations is underway.</p><ul><li>sample checklist (link)</li><li>sample specimen record dataset (link)</li></ul><h3 id="batch-imports" tabindex="-1"><a class="header-anchor" href="#batch-imports" aria-hidden="true">#</a> Batch imports</h3><p>There are various batch importers available within the UI (user interface). These are polished to differing degrees and have various benefits and limitations. The required format is described in the UI. All batch loaders are two-step, allowing for (and requiring) a preview of results before inserting them into the database.</p><ul><li>To explore available batch loaders click on a Data card in the Hub. If batch loader(s) are available then then the batch load link will be enabled.</li><li>Batch importers largely target tab-separated text files, though this is not exclusively the case.</li><li>Notable batch loaders are found in the TaxonNames, Otus, and Sources data cards, though others exist.</li><li>Explore various batch loaders (each data card highlighted in yellow has associated batch loaders at this writing).</li></ul><h4 id="try-batch" tabindex="-1"><a class="header-anchor" href="#try-batch" aria-hidden="true">#</a> Try batch</h4><p>In your test project,</p><ol><li>Go to the data tab</li><li>Select the Otu Data card</li><li>Click “batch load”</li><li>See instructions in the UI for expected / accepted data types and format.</li><li>Create your own file or use this <a href="link">test file</a> Header column = otu_name Blank lines are skipped Tab-delimited format, UTF-8 encoding, Unix line-endings required</li><li>Browse to your file to select it, click preview</li><li>If data looks as expected, browse to select that file again and click create.</li></ol><p>Batch loaders (as of March 2022) include:</p>',10),f={href:"https://docs.taxonworks.org/about/glossary.html#otu-operational-taxonomic-unit",target:"_blank",rel:"noopener noreferrer"},b=e("ul",null,[e("li",null,"simple batch load"),e("li",null,"data attributes"),e("li",null,"simple batch file"),e("li",null,"OTU with identifier batch load")],-1),g=i("<li>collecting events <ul><li>gpx (collecting events with georeferences)</li><li>castor</li></ul></li><li>collection objects <ul><li>castor</li><li>buffered strings</li></ul></li><li>descriptors <ul><li>qualitative descriptors</li><li>modify gene descriptor</li></ul></li><li>sequences <ul><li>Genbank</li><li>Genbank batch</li><li>primers</li></ul></li><li>sources <ul><li>BibTeX</li></ul></li><li>taxon names <ul><li>simple</li><li>castor</li></ul></li><li>asserted distributions <ul><li>simple</li></ul></li><li>namespaces <ul><li>simple</li></ul></li><li>sequence relationships <ul><li>primers batch</li></ul></li>",9),y=i('<h3 id="record-by-record" tabindex="-1"><a class="header-anchor" href="#record-by-record" aria-hidden="true">#</a> Record by record</h3><p>When first learning TaxonWorks, entering records one-at-a-time offers you the opportunity to learn about more of the features in TW and get a feel for how you and others experience the UI.</p><p>For example, you want to enter a specimen record. You have two Tasks enabling you to do this. Choose to use Comprehensive Specimen Digitization Task or the Simple New Specimen Task.</p><h4 id="try-simple-new-specimen" tabindex="-1"><a class="header-anchor" href="#try-simple-new-specimen" aria-hidden="true">#</a> Try Simple New Specimen</h4><p>In your project, try creating a simple new specimen record.</p><ul><li><p>Note you will need to select a namespace. You may find you need to add a namespace before you can do this TW task. Adding a value for namespace ensures your uploaded data records will be unique inside your TW project and across TW projects. In your project, you may also need more than one namespace. [Use Tommy’s INHS Insect Collection as an example, with 12 different namespaces that effectively group the various collections housed at INHS ENT].</p></li><li><p>If you tried the OTU batch loader you can pick one of your OTUs for the name to assign to this specimen.</p></li><li><p>Add an image if you wish</p></li><li><p>Select the Preparation type for this specimen. You may need to add a new value to the dropdown using the New preparation type task.</p></li></ul><h3 id="darwin-core-archive-dwc-a-import" tabindex="-1"><a class="header-anchor" href="#darwin-core-archive-dwc-a-import" aria-hidden="true">#</a> Darwin Core Archive (DwC-A) import</h3><p>To upload occurrence (vouchered specimen data records) data, TW offers the ability to use a DwC Archive file. This provides options to share data that maps to concepts (aka terms, elements, properties) in Darwin Core, but also to specific fields in TaxonWorks and to custom &quot;Data Attribute&quot; fields you can add, if needed.</p><p>For these <strong>occurrence datasets</strong> one needs to map each term to be imported. This next section explains the different mapping options available and steps needed to get started.</p><h3 id="precursor-steps-to-dwc-a-import" tabindex="-1"><a class="header-anchor" href="#precursor-steps-to-dwc-a-import" aria-hidden="true">#</a> Precursor steps to DwC-A import</h3><ul><li><p>For your dataset, you may need to create one or more <strong>namespaces.</strong> These namespaces allow TW to</p><ul><li>track uniqueness of each object and</li><li>group these objects.</li></ul><p>For example, in your collection of physical objects, your catalog numbers may have duplicates if your specimens have come from other instituion collections in the past. Creating namespaces in TW is simple to do, but spending some time on this before upload can simplify the process. Namespaces also need to be unique in your TW Project (first-come, first-served). [NEED EXAMPLES here].</p></li><li><p>In addition, if your dataset has terms not mappable to TW concepts or the DwC terms we currently support, you will need to create custom &quot;Data Attributes.&quot; These need to be created before upload (ideally). If you don&#39;t, then records with these data will not upload but are easy for you to address after other records import.</p></li><li><p>For some <strong>biocuration</strong> data in TW you may find you need to add the group and accepted values for a given class before upload (e.g. group=caste and class=queen).</p></li></ul><h3 id="dwc-terms-mappings" tabindex="-1"><a class="header-anchor" href="#dwc-terms-mappings" aria-hidden="true">#</a> DwC terms mappings</h3><p>Next please note the dwc terms that TW currently recognizes. Also see examples below for how to map to existing TW terms, custom data attributes, and biocuration information.</p><h4 id="record-level-class" tabindex="-1"><a class="header-anchor" href="#record-level-class" aria-hidden="true">#</a> Record-level class</h4><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>type</code></td><td>It is checked that it equals <code>PhysicalObject</code> before allowing the record to be imported. If the value is empty or term not present it is assumed it is a <code>PhysicalObject</code></td></tr><tr><td><code>institutionCode</code></td><td>Selects the repository for the specimen that is registered with an acronym equal to this value</td></tr><tr><td><code>collectionCode</code></td><td>Paired with <code>institutionCode</code> it is used to select the namespace for <code>catalogNumber</code> from a user-defined lookup table in import settings, the value itself is not imported.</td></tr><tr><td><code>basisOfRecord</code></td><td>It is checked that it equals an expected valid value for term, e.g. <code>PreservedSpecimen</code> or <code>FossilSpecimen</code> before allowing the record to be imported. If the value is empty or term not present it is assumed it is a <code>PreservedSpecimen</code> [IS THIS STILL TRUE?]</td></tr></tbody></table><h4 id="occurrence-class" tabindex="-1"><a class="header-anchor" href="#occurrence-class" aria-hidden="true">#</a> Occurrence class</h4><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>catalogNumber</code></td><td>The identifier value for Catalog Number local identifier. The namespace is selected from the namespaces lookup table in import settings queried by <code>institutionCode</code>:<code>collectionCode</code> pair.</td></tr><tr><td><code>recordedBy</code></td><td>It is imported as-is in verbatim collectors field of the collecting event. Additionally, the value is parsed into people and assigned as collectors of the CE. Previously existing people are not used unless the data origin is the same dataset the record belongs to, otherwise any missing people are created.</td></tr><tr><td><code>individualCount</code></td><td>The total number of entities associated with the specimen record (e.g. this record may be for a &quot;lot&quot; containing 6 objects).</td></tr><tr><td><code>sex</code></td><td>Selects the biocuration class from the &quot;sex&quot; biocuration group to be assigned as biocuration classification for the specimen.</td></tr><tr><td><code>preparations</code></td><td>Selects an existing preparation matching the name with this value.</td></tr></tbody></table><h4 id="event-class" tabindex="-1"><a class="header-anchor" href="#event-class" aria-hidden="true">#</a> Event class</h4><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>fieldNumber</code></td><td>Verbatim trip identifier of collecting event</td></tr><tr><td><code>eventDate</code></td><td>The ISO8601-formatted date is split into start year, month and day collecting event fields. If the value is composed of two dates separated by <code>/</code>, then rightmost date is used as end date and split in the same way as start date. If data contradicts dates from other non-empty date-related terms the record will fail to import</td></tr><tr><td><code>eventTime</code></td><td>Time is split into time start hour, minute, and second of collecting event</td></tr><tr><td><code>startDayOfYear</code></td><td>Using <code>year</code> and the value for this term month and day are calculated and stored in start year, month, and day collecting event fields. If the computed value contradicts dates from other non-empty date-related terms the record will fail to import.</td></tr><tr><td><code>endDayOfYear</code></td><td>Using <code>year</code> and the value for this term month and day are calculated and stored in end year, month and day collecting event fields. If the computed value contradicts dates from other non-empty date-related terms the record will fail to import.</td></tr><tr><td><code>year</code></td><td>The start date year of the collecting event. If the value contradicts dates from other non-empty date-related terms the record will fail to import</td></tr><tr><td><code>month</code></td><td>The start date month of the collecting event. If the value contradicts dates from other non-empty date-related terms the record will fail to import.</td></tr><tr><td><code>day</code></td><td>The start date day of the collecting event. If the value contradicts dates from other non-empty date-related terms the record will fail to import</td></tr><tr><td><code>verbatimEventDate</code></td><td>Verbatim date of the collecting event</td></tr><tr><td><code>habitat</code></td><td>Verbatim habitat of the collecting event</td></tr><tr><td><code>samplingProtocol</code></td><td>Verbatim method of the collecting event</td></tr><tr><td><code>fieldNotes</code></td><td>Field notes of the collecting event</td></tr></tbody></table><h4 id="location-class" tabindex="-1"><a class="header-anchor" href="#location-class" aria-hidden="true">#</a> Location class</h4><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>fieldNumber</code></td><td>Verbatim trip identifier of collecting event</td></tr></tbody></table><h4 id="identifcation-class" tabindex="-1"><a class="header-anchor" href="#identifcation-class" aria-hidden="true">#</a> Identifcation class</h4>',22),w=e("thead",null,[e("tr",null,[e("th",null,"Term"),e("th",null,"Mapping")])],-1),v=e("tr",null,[e("td",null,[e("code",null,"identifiedBy")]),e("td",null,[t("A list (concatenated and separated) of names of people, groups, or organizations who assigned the Taxon to the subject. If possible, separate the values in a list with space vertical bar space | (known as a pipe). (e.g. "),e("code",null,"Theodore Pappenfuss | Robert Macey"),t(")")])],-1),x=e("td",null,[e("code",null,"dateIdentified")],-1),k={href:"https://dwc.tdwg.org/terms/#dwc:dateIdentified",target:"_blank",rel:"noopener noreferrer"},_=i('<h4 id="taxon-class" tabindex="-1"><a class="header-anchor" href="#taxon-class" aria-hidden="true">#</a> Taxon class</h4><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>nomenclaturalCode</code></td><td>Selects the nomenclatural code for the taxon ranks used when creating protonyms. The value itself is not imported</td></tr><tr><td><code>kingdom</code></td><td>Creates (unless already present) a protonym at kingdom rank</td></tr><tr><td><code>phylum</code></td><td>Creates (unless already present) a protonym at phylum rank</td></tr><tr><td><code>class</code></td><td>Creates (unless already present) a protonym at class rank</td></tr><tr><td><code>order</code></td><td>Creates (unless already present) a protonym at order rank</td></tr><tr><td><code>family</code></td><td>Creates (unless already present) a protonym at family rank</td></tr><tr><td><code>genus</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>subgenus</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>specificEpithet</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>infraspecificEpithet</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>scientificName</code></td><td>Several protonyms created (only when not present already) with their corresponding ranks and placements</td></tr><tr><td><code>taxonRank</code></td><td>The taxon rank of the most specific protonym</td></tr><tr><td><code>higherClassification</code></td><td>Several protonyms created (only when not present already) with their corresponding ranks and placement. In case a protonym was not already present, only family-group names will be created, names with classsification higher than family-group not previously registered will result in error. Names at genus rank or lower are ignored and extracted from <code>scientificName</code> instead</td></tr><tr><td><code>scientificNameAuthorship</code></td><td>Verbatim author of most specific protonym</td></tr></tbody></table><h3 id="tw-non-standard-mappings" tabindex="-1"><a class="header-anchor" href="#tw-non-standard-mappings" aria-hidden="true">#</a> TW non-standard mappings</h3><p>The DwC importer task includes some TW-specific mappings that are neither DwC core terms nor in any DwC extension term lists but instead, direct mappings to predicates in your projects imported as data attributes for collection objects and collecting events, biocuration groups and classes, and as an advanced-use feature you may have direct mappings to model fields.</p><p><strong>IMPORTANT</strong>: If submitting an actual DwC-A zip file and not tab-separated text file or spreadsheet, this TW-specific mappings have to be placed as headers in the core table, and not in meta.xml. If you are replacing a mapping from meta.xml, you must make sure to comment it out and also if inserting colums make sure you do the appropriate adjustments to avoid collision.</p><h4 id="mappings-to-project-predicates" tabindex="-1"><a class="header-anchor" href="#mappings-to-project-predicates" aria-hidden="true">#</a> Mappings to project predicates</h4><p>In cases where you need to import predicate values targetting the imported collection object or collecting event you may do so by naming the column with a pattern like <code>TW:DataAttribute:&lt;target_class&gt;:&lt;predicate_identifier&gt;</code>. <code>&lt;target_class&gt;</code> may be <code>CollectionObject</code> or <code>CollectingEvent</code>, and the <code>&lt;predicate_identifier&gt;</code> may be the either the name of the predicate or its URI. As an example if you have a predicate registered with name <code>ageInDays</code> and URI <code>http://rs.gbif.org/terms/1.0/ageInDays</code>, both <code>TW:DataAttribute:CollectionObject:ageInDays</code> and <code>TW:DataAttribute:CollectionObject:http://rs.gbif.org/terms/1.0/ageInDays</code> can be used to refer to the same predicate.</p><h4 id="mappings-to-biocuration-groups-and-classes" tabindex="-1"><a class="header-anchor" href="#mappings-to-biocuration-groups-and-classes" aria-hidden="true">#</a> Mappings to biocuration groups and classes</h4><p>The importer is able to map <code>sex</code> into the appropriate biocuration group and select the approriate class according to the value. For additional mappings you may use a special column name pattern to select a biocuration group like <code>TW::BiocurationGroup:&lt;group_identifier&gt;</code> where <code>&lt;group_identifier&gt;</code> can be the name of the biocuration group or its URI. In addition the values must match an existing biocuration class and you may use either its name or URI. For example, if you have a biocuration group registered with name <code>Caste</code> and URI <code>urn:example:ants:caste</code> and biocuration class with name <code>Queen</code> and URI <code>urn:example:ants:caste:queen</code> the following examples do all create the same biocuration classification:</p><table><thead><tr><th>Caste</th><th>urn:example:ants:caste</th></tr></thead><tbody><tr><td>Queen</td><td>urn:example:ants:caste:queen</td></tr><tr><td>urn:example:ants:caste:queen</td><td>Queen</td></tr></tbody></table><h4 id="mappings-to-dwc-predicates" tabindex="-1"><a class="header-anchor" href="#mappings-to-dwc-predicates" aria-hidden="true">#</a> Mappings to DwC predicates</h4><p>Whenever the importer sees that your project has custom attributes for collecting events and/or collection objects that matches Darwin Core URI terms (<code>http://rs.tdwg.org/dwc/terms/&lt;term&gt;</code>), them will be imported as data attributes regardless of any existing mapping of the same field. This allows to preserve verbatim dataaset value for reference as also to import data from terms not supported by the importer.</p><h4 id="direct-mapping-to-tw-model-fields-advanced" tabindex="-1"><a class="header-anchor" href="#direct-mapping-to-tw-model-fields-advanced" aria-hidden="true">#</a> Direct mapping to TW model fields (advanced)</h4>',13),T=e("code",null,"TW:<model_class>:<field>",-1),C={href:"https://docs.taxonworks.org/develop/Data/models.html#collection-object",target:"_blank",rel:"noopener noreferrer"},D=e("code",null,"CollectionObject",-1),I={href:"https://docs.taxonworks.org/develop/Data/models.html#collecting-event",target:"_blank",rel:"noopener noreferrer"},S=e("code",null,"CollectingEvent",-1),j=e("code",null,"<field>",-1),A=i('<table><thead><tr><th>Class</th><th>fields</th></tr></thead><tbody><tr><td><code>CollectionObject</code></td><td><code>buffered_collecting_event</code>, <code>buffered_determinations</code>, <code>buffered_other_labels</code>, <code>total</code>,</td></tr><tr><td><code>CollectingEvent</code></td><td><code>document_label</code>, <code>print_label</code>, <code>verbatim_label</code>, <code>field_notes</code>, <code>formation</code>, <code>group</code>, <code>lithology</code>, <code>max_ma</code>, <code>maximum_elevation</code>, <code>member</code>, <code>min_ma</code>, <code>minimum_elevation</code>, <code>elevation_precision</code>, <code>start_date_day</code>, <code>start_date_month</code>, <code>start_date_year</code>, <code>end_date_day</code>, <code>end_date_month</code>, <code>end_date_year</code>, <code>time_end_hour</code>, <code>time_end_minute</code>, <code>time_end_second</code>, <code>time_start_hour</code>, <code>time_start_minute</code>, <code>time_start_second</code>, <code>verbatim_collectors</code>, <code>verbatim_date</code>, <code>verbatim_datum</code>, <code>verbatim_elevation</code>, <code>verbatim_geolocation_uncertainty</code>, <code>verbatim_habitat</code>, <code>verbatim_latitude</code>, <code>verbatim_locality</code>, <code>verbatim_longitude</code>, <code>verbatim_method</code>, <code>verbatim_trip_identifier</code></td></tr></tbody></table><h3 id="coming-from-other-software" tabindex="-1"><a class="header-anchor" href="#coming-from-other-software" aria-hidden="true">#</a> Coming from other software</h3><h4 id="scratchpads" tabindex="-1"><a class="header-anchor" href="#scratchpads" aria-hidden="true">#</a> Scratchpads</h4><p>We are in the process of exploring two routes to come from Scratchpads to TaxonWorks.</p><ul><li>The DwC import should work well for occurrence data that is based on collected objects.</li><li>The SFG team is has worked with a select number of individual Scratchpad curators to script the process of transferring their datadata. Contact us if you are interested in what this approach entails. Note that this process takes programming effort that is a limited resource within the SFG.</li></ul><h2 id="export" tabindex="-1"><a class="header-anchor" href="#export" aria-hidden="true">#</a> Export</h2><h3 id="full-database-dumps" tabindex="-1"><a class="header-anchor" href="#full-database-dumps" aria-hidden="true">#</a> Full database dumps</h3><p>You can export a full copy of your database, minus binaries (e.g. images, documents) into PostgreSQL format via the <code>Export project database</code> task. This lets you:</p><ul><li>Archive your data</li><li>Load it into a local version of TaxonWorks (e.g. running on your desktop)</li></ul><h3 id="individual-taxonworks-tables" tabindex="-1"><a class="header-anchor" href="#individual-taxonworks-tables" aria-hidden="true">#</a> Individual TaxonWorks tables</h3><p><em>Individual tables in native TaxonWorks format as CSV</em></p><h4 id="via-project-data-overview-and-download" tabindex="-1"><a class="header-anchor" href="#via-project-data-overview-and-download" aria-hidden="true">#</a> Via &quot;Project data overview and download&quot;</h4><ul><li>Open the <code>Project data overview and download</code> task</li><li>Click a <code>Download</code> link corresponding to the table you want to download (22+ tables available)</li></ul><h4 id="via-individual-data-cards" tabindex="-1"><a class="header-anchor" href="#via-individual-data-cards" aria-hidden="true">#</a> Via individual Data cards</h4><ul><li>Click on a Data Card, e.g. <code>Otus</code>.</li><li>On the bar in the middle click on <code>Download</code>, you&#39;ll get a CSV file with the data for that table.</li></ul><h3 id="csv" tabindex="-1"><a class="header-anchor" href="#csv" aria-hidden="true">#</a> CSV</h3><h4 id="basic" tabindex="-1"><a class="header-anchor" href="#basic" aria-hidden="true">#</a> Basic</h4><ul><li><em>Note that these formats are &quot;in progress&quot;, they need refinement.</em></li><li>Open any of the <code>Filter</code> tasks</li><li>Complete a search</li><li>Select the <code>CSV</code> option in the download select and click the blue download button beside it.</li></ul><h3 id="darwin-core-archives" tabindex="-1"><a class="header-anchor" href="#darwin-core-archives" aria-hidden="true">#</a> Darwin Core Archives</h3><p>A quite common data package used for sharing data with biodiversity data aggregators like the Global Biodiversity Information Facility (GBIF) and iDigBio, a <strong>Darwin Core Archive (dwca)</strong> contains several files (e.g. your data file, a metadata file, and an index file to tell computers reading the package what to expect to find in this specific data package). In TaxonWorks, you can create these packages as needed. To do so, you will use the DwC Dashboard shown next.</p><figure><img src="https://sfg.taxonworks.org/s/q4m8is" alt="" title="The DwC Dashboard Task"><figcaption class="figure-left"><strong>Legend</strong>: The <code>DwC Dashboard</code> Task</figcaption></figure><h4 id="from-the-dwc-dashboard" tabindex="-1"><a class="header-anchor" href="#from-the-dwc-dashboard" aria-hidden="true">#</a> From the DwC Dashboard</h4><ul><li>Open the <code>DwC Dashboard</code> task.</li><li>Click one of the green buttons in the <code>Download Darwin Core Archive</code> section. <ul><li>Note your green button options for your dwca file. You may create a dwca for all your data, or a given desired or relevant subset. You can also create your own custom archive (see the <code>Create DwC Archive by filtered collection object result</code> option).</li><li>Once you click on an option, generating the file can sometimes take several seconds to create/load, so wait a few seconds, and you will see a result in the next step.<br><img src="https://sfg.taxonworks.org/s/6nga85" alt="Legend: The TaxonWorks Download Darwin Core Archive panel"></li></ul></li><li>Your download will be queued and built, then available under the <code>Recently created DwC Archives</code> section. <img src="https://sfg.taxonworks.org/s/exphrc" alt="Legend: Recently created DwC Archives"></li></ul><h4 id="from-the-filter-collection-objects-task" tabindex="-1"><a class="header-anchor" href="#from-the-filter-collection-objects-task" aria-hidden="true">#</a> From the <code>Filter collection objects</code> task</h4><ul><li>Open the task <code>Filter collection objects</code>.</li><li>Perform a query to create your desired data subset.</li><li>Use the download select to choose <code>DwC</code>.</li></ul><figure><img src="https://sfg.taxonworks.org/s/8f6ay6" alt="" title="Where to select the dwca format for your `collection object` data download"><figcaption class="figure-left"><strong>Legend</strong>:From <code>Filter Collection Object</code>, select the <code>DwC</code> download format for a given dataset</figcaption></figure><ul><li>Click the blue download button beside the select.</li><li>Optionally click to include data-predicate (custom) fields, scroll down, then click <code>Download</code>.</li></ul><figure><img src="https://sfg.taxonworks.org/s/j450th" alt="" title="Optional fields you can include in your dwca file"><figcaption class="figure-left"><strong>Legend</strong>: Optional fields you can include in your dwca file</figcaption></figure><ul><li>Your download is available on the <code>DwC Dashboard</code> task after a short period.</li></ul><h4 id="preparing-your-data-for-an-aggregator-e-g-gbif-idigbio-ala" tabindex="-1"><a class="header-anchor" href="#preparing-your-data-for-an-aggregator-e-g-gbif-idigbio-ala" aria-hidden="true">#</a> Preparing your data for an aggregator (e.g. GBIF, iDigBio, ALA)</h4>',30),O={href:"https://ipt.gbif.org/manual/en/ipt/latest/dwca-guide",target:"_blank",rel:"noopener noreferrer"},N=e("ol",null,[e("li",null,"data.csv (a tab-separated file (TSV) file that can be opened by most spreadsheet programs). This is your downloaded specimen occurrence data."),e("li",null,"eml.xml (a metadata file with information about your dataset)"),e("li",null,"meta.xml (a metadata file that describes the headers included in your dataset).")],-1),q={href:"https://www.dropbox.com/s/zq4honogcjn64bc/eml_sample.xml?dl=0",target:"_blank",rel:"noopener noreferrer"},W=e("li",null,"Once you've edited this EML file, make sure to re-package it into a ZIP file.",-1),U=e("li",null,"CAUTION, if using a Mac, the system will add two invisible system files when re-zipping. A workaround is available here: https://perishablepress.com/remove-macosx-ds-store-zip-files-mac/",-1),P={href:"https://www.idigbio.org/wiki/index.php/Data_Sharing_Data_Standards_and_Demystifying_the_IPT",target:"_blank",rel:"noopener noreferrer"},E={href:"https://www.gbif.org/ipt",target:"_blank",rel:"noopener noreferrer"},F=i('<h3 id="bibliographies-reference-lists" tabindex="-1"><a class="header-anchor" href="#bibliographies-reference-lists" aria-hidden="true">#</a> Bibliographies (Reference lists)</h3><ul><li>Open the task <code>Filter sources</code></li><li>Use the facets to filter a list of Sources</li></ul><h4 id="as-csv" tabindex="-1"><a class="header-anchor" href="#as-csv" aria-hidden="true">#</a> As CSV</h4><ul><li>Select the <code>CSV</code> option in the download select and click the blue download button beside it.</li></ul><h4 id="as-bibtex" tabindex="-1"><a class="header-anchor" href="#as-bibtex" aria-hidden="true">#</a> As BibTeX</h4><ul><li>Select the <code>BiBTeX</code> option in the download select and click the blue download button beside it.</li></ul><h4 id="formatted-for-journals" tabindex="-1"><a class="header-anchor" href="#formatted-for-journals" aria-hidden="true">#</a> Formatted for journals</h4><ul><li>Select the <code>Download formatted</code> option in the download select and click the blue download button beside it.</li></ul><h5 id="create-shareable-link-to-your-filtered-bibliography" tabindex="-1"><a class="header-anchor" href="#create-shareable-link-to-your-filtered-bibliography" aria-hidden="true">#</a> Create shareable link to your filtered bibliography</h5><p>You can share a zipped copy of the file by generating a link to it. This functionality is turned on if the project administrator has created an API token (See addendum).</p><ul><li>Click <code>Generate download</code></li><li>Copy the link generated and share it.</li></ul><p>A copy of the Download file is also available for future reference, for a period of 2 days, via the <code>Download</code> data card.</p><h4 id="catalog-of-life-data-package-coldp" tabindex="-1"><a class="header-anchor" href="#catalog-of-life-data-package-coldp" aria-hidden="true">#</a> Catalog of Life Data Package (ColDP)</h4><ul><li>Open the task <code>Catalogue of Life (ColDP) exports</code></li><li>Use the <code>OTU</code> autocomplete to choose an OTU. All children of this OTU, by proxy of the attached TaxonName name will be exported.</li><li>Click <code>Download</code></li></ul><p>A copy of the Download file is also available for future reference, for a period of 2 days, via the <code>Download</code> data card.</p><h3 id="data-matrices-evolutionary-descriptive-quantitative-etc" tabindex="-1"><a class="header-anchor" href="#data-matrices-evolutionary-descriptive-quantitative-etc" aria-hidden="true">#</a> Data matrices (evolutionary, descriptive, quantitative, etc.)</h3><ul><li>Open the <code>Obervation matrix hub</code> task</li><li>Click <code>View</code> beside the matrix to be downloaded</li><li>Click the link corresponding to the format you want to download (e.g. <code>TNT</code>, <code>nexus</code>, or <code>NeXML</code>)</li></ul><p>Descriptor lists and OTU contents are also available as downloads here.</p><h4 id="nomenclature-stats" tabindex="-1"><a class="header-anchor" href="#nomenclature-stats" aria-hidden="true">#</a> Nomenclature stats</h4><ul><li>Use the <code>Search a taxon name</code> autocomplete to select a taxon name</li><li>Refine the columns you wish to see and click <code>Search</code> again to update</li><li>Click the blue download icon on the top right</li></ul><h3 id="json" tabindex="-1"><a class="header-anchor" href="#json" aria-hidden="true">#</a> JSON</h3><h4 id="individual-records" tabindex="-1"><a class="header-anchor" href="#individual-records" aria-hidden="true">#</a> Individual records</h4><ul><li>Open a record in &#39;Show&#39;</li><li>Notice the URL format, like <code>/otus/123</code>. Nearly all records can be viewed in with a URL pattern like this (this is a &quot;RESTful&quot; format)</li><li>Add <code>.json</code> to the request, like <code>/otus/123.json</code></li><li>Select all to copy-paste, or use your web-browsers save as to save as a JSON document</li></ul><h4 id="via-filters" tabindex="-1"><a class="header-anchor" href="#via-filters" aria-hidden="true">#</a> Via filters</h4><p><em>All filters can all be used to generate a query that returns JSON</em></p><ul><li>Open a filter</li><li>Complete a search</li><li>Click on the eyeball button and then <code>Show JSON request</code></li><li>Copy the path of the request to your clipboard</li><li>Paste the path of the request to your TaxonWorks base URL (e.g. add it to something like <code>https://my.taxonworks.org/)</code> and hit enter</li><li>Select all to copy-paste, or use your web-browsers save as to save as a JSON document</li></ul><h3 id="the-taxonworks-api" tabindex="-1"><a class="header-anchor" href="#the-taxonworks-api" aria-hidden="true">#</a> The TaxonWorks API</h3>',27),B={href:"https://api.taxonworks.org",target:"_blank",rel:"noopener noreferrer"};function R(L,V){const a=d("ExternalLinkIcon");return n(),s("div",null,[c,h,p,e("p",null,[t("TaxonWorks offers two ways to import data: 1) via the User Interface (UI) entering one-to-many records at a time or batch uploading files, and 2) programmatically with customized scripts for large datasets (e.g. > 1000 records). We seek to make the data import and entry as savvy as possible. As you experience these options, please contribute your ideas for ways to enhance these processes. (Submit ideas, submit code, and please bring your voice to our "),e("a",u,[t("sfgcommons meetings on Wednesdays"),o(a)]),t(").")]),m,e("ul",null,[e("li",null,[t("OTUs "),e("a",f,[t("operational taxonomic units"),o(a)]),b]),g]),y,e("table",null,[w,e("tbody",null,[v,e("tr",null,[x,e("td",null,[t("The date on which the subject was determined as representing the Taxon. Best practice is to use a date that conforms to ISO 8601-1:2019 "),e("a",k,[t("see examples"),o(a)]),t(".")])])])]),_,e("p",null,[t("This is an advance mapping and requires knowledge of the underlying TW models. The pattern is "),T,t(" where model can be either "),e("a",C,[D,o(a)]),t(" or "),e("a",I,[S,o(a)]),t(", and "),j,t(" can be the ones listed below.")]),A,e("ul",null,[e("li",null,[t("Your export will download as a "),e("a",O,[t("DWCA"),o(a)]),t(", that includes three files (or more) when unzipped: "),N]),e("li",null,[t("You can (and will need to) edit your eml.xml file in a plain text editor to prepare it for upload to an aggregator. A sample file that works is available "),e("a",q,[t("here"),o(a)]),t(".")]),W,U,e("li",null,[t("Once repackaged, add this zipped file to your Integrated Publishing Toolkit (IPT) endpoint. Note there are existing IPT instances that would be happy to host your packaged data. More information on setting one up here: "),e("ol",null,[e("li",null,[e("a",P,[t("iDigBio"),o(a)])]),e("li",null,[e("a",E,[t("GBIF"),o(a)])])])])]),F,e("p",null,[t("See "),e("a",B,[t("api.taxonworks.org"),o(a)]),t(".")])])}const Y=r(l,[["render",R],["__file","import-and-export.html.vue"]]);export{Y as default};
